examples/file-server/src/test/file-server.ix.ts

```diff
@@ -41,11 +42,14 @@ describe('File Server Feature', function () {
 
     it('validating metrics are being cleared between runs', async () => {
         const { getMetrics } = await getLoadedFeature();
-        const metrics = await getMetrics();
-        expect(metrics.marks.length).to.eq(2);
-        const browserMark = metrics.marks.find((m) => m.name === MAIN_MARK);
-        const serverMark = metrics.marks.find((m) => m.name === SERVER_MARK);
-        expect(browserMark, 'should receive browser marks').not.be.undefined;
-        expect(serverMark, 'should receive server marks').not.be.undefined;
+
+        await waitFor(async () => {
+            const metrics = await getMetrics();
```

Comment:
how many of these changes repos down the line would have?

Reply:
Depending on the assumptions used in tests.
With the latest revision, moving `navigationOptions` to the options object passed to `withFeature/getLoadedFeature`, this test would require an even smaller change: `navigationOptions: { waitUntil: 'networkidle' }` to be passed in. That would reproduce the previous behavior.

The thing is... network being idle doesn't _necessarily_ say all code executed and finished its work (or that metrics are ready). I left the `waitFor` in, as I feel it would make the test a bit more robust and harder to break.